src/apprt/gtk/Window.zig

```diff
@@ -407,19 +365,16 @@ pub fn syncAppearance(self: *Window, config: *const configpkg.Config) !void {
     // Disable the title buttons (close, maximize, minimize, ...)
     // *inside* the tab overview if CSDs are disabled.
     // We do spare the search button, though.
-    if ((comptime adwaita.versionAtLeast(1, 4, 0)) and
-        adwaita.enabled(&self.app.config))
-    {
-        if (self.tab_overview) |tab_overview| {
-            c.adw_tab_overview_set_show_start_title_buttons(
-                @ptrCast(tab_overview),
-                @intFromBool(csd_enabled),
-            );
-            c.adw_tab_overview_set_show_end_title_buttons(
-                @ptrCast(tab_overview),
-                @intFromBool(csd_enabled),
-            );
-        }
+    if (self.tab_overview) |tab_overview| {
+        if (!adwaita.versionAtLeast(1, 4, 0)) unreachable;
```

Comment:
Should you move this if outside the other one?

Reply:
Also as a note `if (x) unreachable` is equivalent to `assert(!x)`, so we should just use an assert.

Reply:
switched to `assert()` where appropriate.